projects/plugins/jetpack/extensions/store/membership-products/resolvers.js

```diff
@@ -22,12 +22,18 @@ import { API_STATE_CONNECTED, API_STATE_NOTCONNECTED } from './constants';
 import getConnectUrl from '../../shared/get-connect-url';
 import { PRODUCT_TYPE_PAYMENT_PLAN } from '../../shared/components/product-management-controls/constants';
 import { getMessageByProductType } from '../../shared/components/product-management-controls/utils';
+import executionLock from '../../shared/execution-lock';
+
+const EXECUTION_KEY = 'membership-products-resolver-getProducts';
 
 export const getProducts = (
```

Comment:
This function is very complicated to test in its current state, that's why I haven't added tests for it.